q2cli/builtin/tools.py

```diff
@@ -907,3 +908,249 @@ def cache_status(cache):
+@click.option('--suppress-header/--no-suppress-header',
+              default=False,
+              show_default=True,
+              help='do not write header/footer blocks in the output script')
+@click.option('--verbose/--no-verbose',
+              default=True,
+              show_default=True,
+              help='print status messages to stdout while processing')
+@click.option('--dump-recorded-metadata/--no-dump-recorded-metadata',
+              default=True,
+              show_default=True,
+              help='write the original metadata captured in provenance to '
+                   'disk in the --metadata-out-fp directory')
+@click.option('--metadata-out-fp',
+              default='',
+              show_default=True,
+              help=('the directory where captured study metadata '
+                    'should be written if --dump-recorded-metadata. This '
+                    'often produces many outputs, so a dedicated directory '
+                    'should generally be used. Creates the directory if it '
+                    'does not already exist. By default, metadata is written '
+                    'to `${PWD}/recorded_metadata/`'))
+@click.option('--out-fp',
+              required=True,
+              help='the filepath where your replay script should be written.')
+def provenance_replay(
+    in_fp: str,
+    out_fp: str,
+    usage_driver: Literal['python3', 'cli'],
+    recurse: bool = False,
+    validate_checksums: bool = True,
+    parse_metadata: bool = True,
+    use_recorded_metadata: bool = False,
+    suppress_header: bool = False,
+    verbose: bool = True,
+    dump_recorded_metadata: bool = True,
+    metadata_out_fp: str = ''
+):
+    """
+    Replay provenance from a QIIME 2 Artifact filepath to a written executable
+    """
+    from qiime2.core.archive.provenance_lib.replay import replay_provenance
+    from qiime2.sdk.util import get_available_usage_drivers
+
+    usage_drivers = get_available_usage_drivers()
+    try:
+        usage_driver_type = usage_drivers[usage_driver]
+    except KeyError:
+        msg = (
+            f'The {usage_driver} usage driver is not available in the '
+            'current evnironment.'
+        )
+        raise ValueError(msg)
+
+    replay_provenance(
+        usage_driver=usage_driver_type,
+        payload=in_fp,
+        out_fp=out_fp,
+        validate_checksums=validate_checksums,
+        parse_metadata=parse_metadata,
+        recurse=recurse,
+        use_recorded_metadata=use_recorded_metadata,
+        suppress_header=suppress_header,
+        verbose=verbose,
+        dump_recorded_metadata=dump_recorded_metadata,
+        md_out_fp=metadata_out_fp
+    )
+    filename = os.path.realpath(out_fp)
+    click.echo(f'{usage_driver} replay script written to {filename}')
+
+
+@tools.command(name='replay-citations', cls=ToolCommand)
+@click.option('--in-fp', required=True,
+              help='filepath to a QIIME 2 Archive or directory of Archives')
```

Comment:
I recommend assigning the `help` text that you're re-using for options that show up in multiple commands to variables, and then providing those variables when defining options. That would let my suggestions above (and any future edits) propagate to all commands that use those options.